### PR TITLE
wait-for-migration should output info to log before calling API's

### PR DESCRIPTION
**Pull request by mulana**
closes: https://github.com/github/gh-gei/issues/683

Adding a info log before API call, as sometimes logs weird if an API call fails.

Screenshot 2023-05-25 at 1 45 31 PM

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

**Comment by hfishback01** [comment]
Looks great, @mulana ! Can you make sure to update the release notes in this PR? I think we're supposed to do that with each PR.

You can find them [here](https://github.com/github/gh-gei/blob/main/RELEASENOTES.md). You can just add a bullet point to the end of the list with a customer-facing description of what your update is. 

Maybe something like "Updated log order during migration to better affect readability"?

File: src/Octoshift/Commands/WaitForMigration/WaitForMigrationCommandHandler.cs [hunk 1]
```diff
@@ -75,9 +75,11 @@ private async Task WaitForOrgMigration(string migrationId, GithubApi githubApi)
 
     private async Task WaitForRepositoryMigration(string migrationId, GithubApi githubApi)
     {
+        _log.LogInformation($"Waiting for migration (ID: {migrationId}) to finish...");
+
         var (state, repositoryName, failureReason, migrationLogUrl) = await githubApi.GetMigration(migrationId);
 
-        _log.LogInformation($"Waiting for {repositoryName} migration (ID: {migrationId}) to finish...");
+        _log.LogInformation($"Waiting for migration of repository (name: {repositoryName}) to finish...");
```
Comment (synthead) [inline src/Octoshift/Commands/WaitForMigration/WaitForMigrationCommandHandler.cs:82]: Small :pinching_hand: phrasing nit :grin: 

```suggestion
        _log.LogInformation($"Waiting for migration of repository {repositoryName} to finish...");
```

**Comment by mulana** [comment]
@hfishback01 and @synthead I have applied requested changes 👍🏼 . When you have time can you take another look? 👀

**Review by hfishback01 — APPROVED** [review]
LGTM! 🎉  

Let me be the first to say: way to go on first c# 🚀 !

File: RELEASENOTES.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
 - Add new '--skip-invitation' flag for `reclaim-mannequin` to allow EMU organizations to reclaim mannequins without an email invitation
 - Write warnings to log and console if GitHub is experiencing an availability incident.
 - Improve the error thrown when you have insufficient permissions for the target GitHub organization to explicitly mention the relevant organization
+- Updated log order during migration for better readability
```
Comment (synthead) [inline RELEASENOTES.md:4]: How about:

```suggestion
- Write log output prior to making API calls in wait-for-migration commands
```

**Review by hfishback01 — APPROVED** [review]

**Review by synthead — APPROVED** [review]
Looks great!  Congrats on the first CLI ship, @mulana! :heart:

**Pull request merged**